test/runtests.jl

```diff
@@ -500,3 +500,56 @@ end
     pa2 = repeat(pa1, inner = (2, 1))
     @test pa2 == [1 2; 1 2; 3 4; 3 4]
 end
+
+@testset "map pure tests" begin
```

Comment:
Can you add a few `@inferred` checks where applicable?

Reply:
I add the `@inferred` tests elsewhere, as here nothing is inferred since now inference produces the `Union` of two concrete types, see:
```
julia> x = PooledArray(fill(1, 200), signed=true, compress=true);

julia> @inferred map(Int, x)
ERROR: return type PooledVector{Int64, Int8, Vector{Int8}} does not match inferred return type Union{PooledVector{Int64, 
Int64, Vector{Int64}}, PooledVector{Int64, Int8, Vector{Int8}}}

julia> @inferred map(Int, x, pure=true)
ERROR: return type PooledVector{Int64, Int8, Vector{Int8}} does not match inferred return type Union{PooledVector{Int64, 
Int64, Vector{Int64}}, PooledVector{Int64, Int8, Vector{Int8}}}
```

Reply:
I added `@inferred` with `Union` for `map` tests